lib/modules/landing/landing_presenter.dart

```diff
@@ -5,6 +5,19 @@ abstract class LandingPresenter
 
 class LandingPresenterImpl extends LandingPresenter
     implements LandingInteractorDelegate, LandingViewDelegate {
+  LandingPresenterImpl() {
+    stateDidInit.voidListen(_stateDidInit).addTo(disposeBag);
+    alertDialogDidClose.voidListen(_alertDialogDidClose).addTo(disposeBag);
+    authenticationDidFailToValidate
+        .listen(_authenticationDidFailToValidate)
+        .addTo(disposeBag);
+
+    animationDidFinish
+        .zipWith<bool, bool>(authenticationDidValidate, (t, s) => s)
```

Comment:
should it be `(t, s) => t && s` ? we're zipping the condition so the result should be combined right?

Reply:
No no, my intent here that we have to wait for `animationDidFinish` & `authenticationDidValidate ` both emit
then take the result from `authenticationDidValidate` and triggers something

Reply:
yeah I understood the idea of zip operator to wait for both. But then in this case animationDidFinish doesn't seem to ever emit false right? 🤔 should we just have a void type instead of bool?

Reply:
I think the [zipper](https://pub.dev/documentation/rxdart/latest/rx/ZipWithExtension/zipWith.html) param is like a mapping func right?  Or I didn't get your point there 😁 

```dart
Stream.fromIterable([1])
    .zipWith(Stream.fromIterable([2]), (one, two) => one + two)
    .listen(print); // prints 3
```

Reply:
lol I meant...shouldn't it be `zipWith<void, bool>` @markgravity ?😅

Reply:
and then since the first args is unused, should we put it as `_`?

Reply:
> lol I meant...shouldn't it be zipWith<void, bool> @markgravity ?😅

It should be the type of second stream 😁 
<img width="1127" alt="Screen Shot 2021-05-18 at 10 50 34" src="https://user-images.githubusercontent.com/17875522/118587483-e4810380-b7c6-11eb-9d24-9af4c7443511.png">

Reply:
hmmm it's kinda weird don't you think? 🤔 because the first stream is emitting `void` as the type 🤔 why does it accept bool in this first place lol